gtda/diagrams/_metrics.py

```diff
@@ -51,204 +65,273 @@ def heats(diagrams, sampling, step_size, sigma):
-        bottleneck_distance(
-            diagram_1[diagram_1[:, 0] != diagram_1[:, 1]],
-            diagram_2[diagram_2[:, 0] != diagram_2[:, 1]], delta)
-        for diagram_2 in diagrams_2] for diagram_1 in diagrams_1])
-
-
-def wasserstein_distances(diagrams_1, diagrams_2, p=2, delta=0.01,
-                          **kwargs):
-    return np.array([[
-        wasserstein_distance(
-            diagram_1[diagram_1[:, 0] != diagram_1[:, 1]],
-            diagram_2[diagram_2[:, 0] != diagram_2[:, 1]], p, delta)
-        for diagram_2 in diagrams_2] for diagram_1 in diagrams_1])
-
-
-def heat_distances(diagrams_1, diagrams_2, sampling, step_size,
-                   sigma=1., p=2., **kwargs):
-    heat_1 = heats(diagrams_1, sampling, step_size, sigma).\
-        reshape(diagrams_1.shape[0], -1)
+    return np.array(
+        [
+            [
+                bottleneck_distance(
+                    diagram_1[diagram_1[:, 0] != diagram_1[:, 1]],
+                    diagram_2[diagram_2[:, 0] != diagram_2[:, 1]],
+                    delta,
+                )
+                for diagram_2 in diagrams_2
+            ]
+            for diagram_1 in diagrams_1
+        ]
+    )
+
+
+def wasserstein_distances(diagrams_1, diagrams_2, p=2, delta=0.01, **kwargs):
+    return np.array(
+        [
+            [
+                wasserstein_distance(
+                    diagram_1[diagram_1[:, 0] != diagram_1[:, 1]],
+                    diagram_2[diagram_2[:, 0] != diagram_2[:, 1]],
+                    p,
+                    delta,
+                )
+                for diagram_2 in diagrams_2
+            ]
+            for diagram_1 in diagrams_1
+        ]
+    )
+
+
+def heat_distances(
+    diagrams_1, diagrams_2, sampling, step_size, sigma=1.0, p=2.0, **kwargs
+):
+    heat_1 = heats(diagrams_1, sampling, step_size, sigma).reshape(
+        diagrams_1.shape[0], -1
+    )
     if np.array_equal(diagrams_1, diagrams_2):
-        unnorm_dist = squareform(pdist(heat_1, 'minkowski', p=p))
+        unnorm_dist = squareform(pdist(heat_1, "minkowski", p=p))
         return (step_size ** (1 / p)) * unnorm_dist
-    heat_2 = heats(diagrams_2, sampling, step_size, sigma).\
-        reshape(diagrams_2.shape[0], -1)
-    unnorm_dist = cdist(heat_1, heat_2, 'minkowski', p=p)
+    heat_2 = heats(diagrams_2, sampling, step_size, sigma).reshape(
+        diagrams_2.shape[0], -1
+    )
+    unnorm_dist = cdist(heat_1, heat_2, "minkowski", p=p)
     return (step_size ** (1 / p)) * unnorm_dist
 
 
-def persistent_image_distances(diagrams_1, diagrams_2, sampling, step_size,
-                               sigma=1., p=2., **kwargs):
-    persistent_image_1 = persistent_images(diagrams_1, sampling,
-                                           step_size, sigma).\
-        reshape(diagrams_1.shape[0], -1)
+def persistent_image_distances(
+    diagrams_1, diagrams_2, sampling, step_size, sigma=1.0, p=2.0,
+    weight_function=lambda x: x[1], **kwargs
```

Comment:
The weight function should not be an optional parameter in this function. All transformers using persistent image have to pass it. The transformer themselves will have default value for weight function.

Reply:
Changed. Does the same apply for sigma?